crates/sui-cluster-test/src/cluster.rs

```diff
@@ -156,12 +186,81 @@ impl Cluster for LocalNewCluster {
         &self.fullnode_url
     }
 
-    // For now, a local cluster does not have faucet
-    fn faucet_url(&self) -> Option<&str> {
+    fn user_key(&self) -> AccountKeyPair {
+        get_key_pair().1
+    }
+
+    fn remote_faucet_url(&self) -> Option<&str> {
         None
     }
 
+    fn local_faucet_key(&self) -> Option<&AccountKeyPair> {
+        Some(&self.faucet_key)
+    }
+}
+
+// Make linter happy
+#[async_trait]
+impl Cluster for Box<dyn Cluster + Send + Sync> {
+    async fn start(_options: &ClusterTestOpt) -> Result<Self, anyhow::Error> {
+        unreachable!(
+            "If we already have a boxed Cluster trait object we wouldn't have to call this function"
+        );
+    }
+    fn rpc_url(&self) -> &str {
+        (**self).rpc_url()
+    }
+
+    fn fullnode_url(&self) -> &str {
+        (**self).fullnode_url()
+    }
+
     fn user_key(&self) -> AccountKeyPair {
-        self.swarm().config().account_keys[0].copy()
+        (**self).user_key()
+    }
+
+    fn remote_faucet_url(&self) -> Option<&str> {
+        (**self).remote_faucet_url()
+    }
+
+    fn local_faucet_key(&self) -> Option<&AccountKeyPair> {
+        (**self).local_faucet_key()
     }
 }
+
+pub async fn new_wallet_context_from_cluster(
+    cluster: &(dyn Cluster + Sync + Send),
+    key_pair: AccountKeyPair,
+) -> WalletContext {
+    let temp_dir = tempfile::tempdir().unwrap();
+    let wallet_config_path = temp_dir.path().join("client.yaml");
+    let rpc_url = cluster.rpc_url();
+    info!("Use gateway: {}", &rpc_url);
+    let keystore_path = temp_dir.path().join(SUI_KEYSTORE_FILENAME);
+    let keystore = KeystoreType::File(keystore_path);
+    let address: SuiAddress = key_pair.public().into();
+    keystore.init().unwrap().add_key(key_pair).unwrap();
+    SuiClientConfig {
+        accounts: vec![address],
+        keystore,
+        gateway: ClientType::RPC(rpc_url.into()),
+        active_address: Some(address),
+    }
+    .persisted(&wallet_config_path)
+    .save()
+    .unwrap();
+
+    info!(
+        "Initialize wallet from config path: {:?}",
+        wallet_config_path
+    );
+
+    WalletContext::new(&wallet_config_path)
+        .await
+        .unwrap_or_else(|e| {
+            panic!(
+                "Failed to init wallet context from path {:?}, error: {e}",
+                wallet_config_path
+            )
+        })
+}
```

Comment:
Nit: you could probably do this with an `expect`?

Reply:
I want string interpolation while with `expect` I need `format!()`